Focus on user impact in relevancy section (#50)
meilisearch/engine-team: meilisearch-release-changelog/v1.2.0.md

### Relevancy improvements

- The `exactness` ranking rule no longer treats synonyms as exact matches. This boosts documents containing the query exactly as typed by the user. Solves https://github.com/meilisearch/meilisearch/issues/3410
- Meilisearch will always sort result as if the `words` ranking rule were present with a higher priority than the `attributes`, `exactness`, `typo` and `proximity` ranking rules, even if in the settings the `words` ranking rule has been removed or set with a lower priority. Search behavior when using the specified ranking rules before or without `words` was confusing and hurt relevancy
- Split words are now also treated as possible digrams. This means the query `whit ehorse` may match documents containing `white horse`, which improves the relevancy of typo tolerance

> - N-grams and split words are now disadvantaged in comparison to the exact word in the `typo` ranking rule. This means that, e.g., a query for `sun flower` will now return a document titled `Sun Flower` before one titled `Sunflower`

Reviewer: Trying to get rid of disadvantaged in favour of something a bit simple if possible—it's a bit of a mouthful. Also, it's usually better to use "for example" instead of "e.g." when writing to a wide audience that includes non-native speakers.

```suggestion
- N-grams and split words are now ranked lower in comparison to the exact word when processing the `typo` ranking rule. For example, a query for `sun flower` will now return a document titled `Sun Flower` before one titled `Sunflower`
```

> - The behavior of the ranking rules no longer depends on how many documents they have to rank, improving consistency in terms of both relevancy and latency. Solves https://github.com/meilisearch/meilisearch/issues/3356

Reviewer: Minor improvements to sentence flow.

```suggestion
- Ranking rule behavior no longer depends on the number of ranked documents, improving consistency in terms of both relevancy and latency. Solves https://github.com/meilisearch/meilisearch/issues/3356
```

- The `exactness` ranking rule no longer treats synonyms as exact matches. This boosts documents containing the query exactly as typed by the user. Solves https://github.com/meilisearch/meilisearch/issues/3410

> - When using the `attributes`, `exactness`, `typo`, and `proximity` ranking rules, Meilisearch will sort results as if  the`words` ranking rule were present and with higher priority. Search behavior when using the specified ranking rules before or without `words` was confusing and hurt relevancy

Reviewer 2: Could we add something like "even if `words` ranking rule has been removed..."
what do you think @guimachiavelli and @dureuill?

Author: Attempting:

```
Meilisearch will always sort result as if the `words` ranking rule were present with a higher priority than the `attributes`, `exactness`, `typo` and `proximity` ranking rules, even if in the settings the `words` ranking rule has been removed or set with a lower priority. Search behavior when using the specified ranking rules before or without `words` was confusing and hurt relevancy
```

Reviewer 2: For me it's perfect 👌

- Split words are now also treated as possible digrams. This means the query `whit ehorse` may match documents containing `white horse`, which improves the relevancy of typo tolerance
- N-grams and split words are now disadvantaged in comparison to the exact word in the `typo` ranking rule. This means that, e.g., a query for `sun flower` will now return a document titled `Sun Flower` before one titled `Sunflower`
- The behavior of the ranking rules no longer depends on how many documents they have to rank, improving consistency in terms of both relevancy and latency. Solves https://github.com/meilisearch/meilisearch/issues/3356

> - Exactness will no longer prioritize documents containing exact words which have already been parsed and removed by the `words` ranking rule. For example, when using the query `Batman: the dark knight rises` with the default term matching strategy (`last`), documents with a title starting with `Batman: the dark knight` will be returned before documents containing additional terms like `Batman: the dark knight crashes`

Author: Removing then

> - Exactness will no longer prioritize documents containing exact words which have already been parsed and removed by the `words` ranking rule. For example, when using the query `Batman: the dark knight rises` with the default term matching strategy (`last`), documents with a title starting with `Batman: the dark knight` will be returned before documents containing additional terms like `Batman: the dark knight crashes`

Author: removed
